File: Programmers/YoungJun/4주차(정렬)/H-index.py
def solution(citations):
    answer = 0
    for n in range(len(citations)+1):#n이 리스트의 갯수까지 될 수 있기에 길이에 +1 추가
        h = 0
        for cit_i in citations: # 이중 for문을 돌려 citations 값이 n이상일 시 h증가 
            if n <= cit_i:
                h += 1
        if n <= h and n-len(citations) <= h:
            # n편 중, 인용된 논문이 h편 이상이고 나머지 논문이 h번 이하일 시 answer값으로 변경
            answer = n
    return answer

File: Programmers/SeungDo/4주차 - 정렬/H-Index.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;
// 오름차순으로 정렬한 후 citations.size만큼 반복문을 돌려 answer에 H_Index값을 넣고 
// citations[i]값이 answer값보다 큰지 비교 후 리턴 
// 문제가 이해가 잘 안돼서 풀이를 보고 진행하였습니다. https://yabmoons.tistory.com/694
int solution(vector<int> citations)
{
	sort(citations.begin(), citations.end()); // 오름차순 정렬
	for (int i  = 0; i  < citations.size(); i++)
	{
		int answer  = citations.size() - i; // h_Index값 넣음
		if (citations[i] >= answer) return answer; // h_Index값보다 인용횟수가 크거나 같다면 h_Index값 리턴
	}
    return 0; // 인용횟수가 전부다 0 일때 0 리턴
}
